dbt/adapters/default.py

```diff
@@ -414,7 +414,7 @@ def release_connection(cls, profile, name):
 
     @classmethod
     def cleanup_connections(cls):
-        global connections_in_use, connections_available
+        global connections_in_use, connections_available, lock
```

Comment:
is this extraneous or does it need to be here?

Reply:
idk, i think you're supposed to declare globals this way. i am not a python expert

Reply:
haha, ok. I think it's meant to be used when you redefine a variable. Otherwise, doing `lock = something_else` will define a new variable, `lock` in the local function scope. The `global` keyword will tell python to overwrite the var in the global scope instead. I don't think it's needed if you're mutating a variable, like adding to a list or a dict